### DISCUSSION TITLE
Instance 'to_world' param inverse problem

### DISCUSSION THREAD

**Discussion by gnauk** [mitsuba-renderer/mitsuba3#715] [Q&A] [answered]
I rerwite the tutorial 'object pose estimate' with the following formula, using the instance and shapegroup . 

```
    'obj': {
       'type': 'shapegroup',
        'bunny': {
          'type': 'ply',
          'filename': './scenes/meshes/bunny.ply',
          'bsdf': {
            'type': 'diffuse',
            'reflectance': { 'type': 'rgb', 'value': (0.3, 0.3, 0.75) },
          }
       }
    },
    'insta1':{
        'type': 'instance',
        'to_world': T.translate([0.2, 0.2, 0]).rotate([0, 1, 0], 45).scale(6.5),
        'shapegroup': {
            'type': 'ref',
            'id': 'obj'
        }
    }
```
and 

```
def apply_transformation(params, opt):
    opt['trans'] = dr.clamp(opt['trans'], -0.5, 0.5)
    opt['angle'] = dr.clamp(opt['angle'], -0.5, 0.5)
    trafo = mi.Transform4f.translate([opt['trans'].x, opt['trans'].y, 0.0]).rotate([0, 1, 0], opt['angle'] * 100.0).scale(6.5)
    params['insta1.to_world'] = trafo
    params.update()
```

The error is as follows:

`Criticla Dr.jit compiler failure: jitc_vcall_upload(): could not find callable!`

Another question, if the target is to optimize the position for mutiple instances, the use of prb_reparam integetor is right?

**Comment by njroussel** [comment]
Hi @gnauk 

I can reproduce this, this is most likely a bug - I'm pretty sure we could optimize instances a while ago.

You are correct, you should use either `prb_reparam` or `direct_reparam` for such a task.

**Reply by gnauk** [reply]
@njroussel  Thanks! When bugs occured with Dr.jit, we can't dissolve. The tortuial to debug with Dr.jit is needed.

**Reply by njroussel** [reply]
This is an issue deep within Dr.Jit. It requires quite a bit of knowledge of DrJit's internals in order to fix it. We can definitely improve some of the documentation, but I doubt we'll ever be able to fit something like this into a tutorial.

I opened a pull request over here: https://github.com/mitsuba-renderer/drjit-core/pull/61
This should fix your issue.

**Reply by njroussel** [reply, marked as answer]
Fixed in Mitsuba with https://github.com/mitsuba-renderer/mitsuba3/commit/9273af4798f27a9235fc275753a2779088fb4bfd
